-Werror with -Wstrict-overflow causes multiple build errors on release builds

##### Brief Description
Commit hash:  ab12555935fe5a39f1cb3470fb2daaed5b68ffc8

Reproducible: Always

##### Steps to reproduce

1. cmake -DCMAKE_BUILD_TYPE=Release .
2. make

##### Observed Behavior
The O2 optimization causes some assumptions by the compiler that then causes warnings. I see multiple warnings like "assuming signed overflow does not occur when assuming that ..." generated from -Werror=strict-overflow, stopping the build.

Some of the errors seem extremely minor, like looping on the size of a QVector which is of signed int type.

A couple examples:
```
[  9%] Building CXX object CMakeFiles/qtox_static.dir/src/core/coreav.cpp.o
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp: In member function ‘QTextDocument* DocumentCache::pop()’:
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp:29:16: error: assuming signed overflow does not occur when assuming that (X - c) <= X is always true [-Werror=strict-overflow]
 QTextDocument* DocumentCache::pop()
                ^
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp: In destructor ‘DocumentCache::~DocumentCache()’:
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp:23:1: error: assuming signed overflow does not occur when assuming that (X - c) <= X is always true [-Werror=strict-overflow]
 DocumentCache::~DocumentCache()
 ^
cc1plus: all warnings being treated as errors
CMakeFiles/qtox_static.dir/build.make:602: recipe for target 'CMakeFiles/qtox_static.dir/src/chatlog/documentcache.cpp.o' failed
```

I see some only on OBS, and different ones for different distros, I guess because of different compiler versions..

here's a union of all the warnings I see:
OBS only:
```
[  316s] /usr/src/packages/BUILD/src/persistence/settingsserializer.cpp: In member function 'void SettingsSerializer::save()':
[  316s] /usr/src/packages/BUILD/src/persistence/settingsserializer.cpp:267:9: warning: assuming signed overflow does not occur when simplifying conditional to constant [-Wstrict-overflow]
[  316s]          if (g != -1) {
[  316s]          ^~
```
local and OBS:
```
[  382s] /usr/src/packages/BUILD/src/platform/posixsignalnotifier.cpp: In function 'void detail::signalHandler(int)':
[  382s] /usr/src/packages/BUILD/src/platform/posixsignalnotifier.cpp:56:60: warning: ignoring return value of 'ssize_t write(int, const void*, size_t)', declared with attribute warn_unused_result [-Wunused-result]
[  382s]      ::write(g_signalSocketPair[0], &signum, sizeof(signum));
[  382s]                                                             ^
[  382s] /usr/src/packages/BUILD/src/platform/posixsignalnotifier.cpp: In member function 'void PosixSignalNotifier::onSignalReceived()':
[  382s] /usr/src/packages/BUILD/src/platform/posixsignalnotifier.cpp:112:67: warning: ignoring return value of 'ssize_t read(int, void*, size_t)', declared with attribute warn_unused_result [-Wunused-result]
[  382s]      ::read(detail::g_signalSocketPair[1], &signum, sizeof(signum));
[  382s]                                                                    ^
```
local only:
```
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp: In member function ‘QTextDocument* DocumentCache::pop()’:
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp:29:16: warning: assuming signed overflow does not occur when assuming that (X - c) <= X is always true [-Wstrict-overflow]
 QTextDocument* DocumentCache::pop()
                ^
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp: In destructor ‘DocumentCache::~DocumentCache()’:
/home/abilinski/build/qTox/src/chatlog/documentcache.cpp:23:1: warning: assuming signed overflow does not occur when assuming that (X - c) <= X is always true [-Wstrict-overflow]
 DocumentCache::~DocumentCache()
 ^
```

--- a/src/platform/posixsignalnotifier.cpp
+++ b/src/platform/posixsignalnotifier.cpp
@@ -53,7 +53,12 @@ static void signalHandler(int signum)
     if (g_signalSocketUsageFlag.test_and_set())
         return;
 
-    ::write(g_signalSocketPair[0], &signum, sizeof(signum));
+    if(::write(g_signalSocketPair[0], &signum, sizeof(signum)) == -1) {
+        // We hardly can do anything more usefull in signal handler, and
+        // any ways it's probably very unexpected error (out of memory?),
+        // since we check socket existance with a flag beforehand.
+        abort();
+    }
 
     g_signalSocketUsageFlag.clear();
 }
@@ -109,7 +114,9 @@ PosixSignalNotifier& PosixSignalNotifier::globalInstance()
 void PosixSignalNotifier::onSignalReceived()
 {
     int signum{0};
-    ::read(detail::g_signalSocketPair[1], &signum, sizeof(signum));
+    if (::read(detail::g_signalSocketPair[1], &signum, sizeof(signum)) == -1) {
+        qFatal("Failed to read from signal socket, error = %d", errno);
+    }
 
     qDebug() << "Signal" << signum << "received";
     emit activated(signum);

